### PR TITLE
RetrieveView mostly fixed

### PR DESCRIPTION
**Pull request by ryotahidaka**
Need to add implementation were exercise.csv gets a new user added every time a new user is created.
See other TODOs.

### CODE REVIEW

File: src/java/app/Main.java [hunk 1]
```diff
@@ -53,7 +53,7 @@ public static void main(String[] args) {
         SignupView signupView = SignupUseCaseFactory.create(viewManagerModel, loginViewModel, signupViewModel, userDataAccessObject);
         views.add(signupView, signupView.viewName);
 
-        LoginView loginView = LoginUseCaseFactory.create(viewManagerModel, loginViewModel, loggedInViewModel, userDataAccessObject, signupViewModel);
+        LoginView loginView = LoginUseCaseFactory.create(viewManagerModel, loginViewModel, loggedInViewModel, userDataAccessObject, signupViewModel, retrieveViewModel);
```
Comment (ryotahidaka) [inline src/java/app/Main.java:56]: Adding retreieve

File: src/java/app/LoginUseCaseFactory.java [hunk 2]
```diff
@@ -7,6 +7,7 @@
 import interface_adapter.login.LoginController;
 import interface_adapter.login.LoginPresenter;
 import interface_adapter.login.LoginViewModel;
+import interface_adapter.retrieve.RetrieveViewModel;
```
Comment (ryotahidaka) [inline src/java/app/LoginUseCaseFactory.java:10]: Adding retrieve

File: src/java/app/LoginUseCaseFactory.java [hunk 3]
```diff
@@ -27,10 +28,11 @@ public static LoginView create(
             LoginViewModel loginViewModel,
             LoggedInViewModel loggedInViewModel,
             LoginUserDataAccessInterface userDataAccessObject,
-            SignupViewModel signupViewModel) {
+            SignupViewModel signupViewModel,
+            RetrieveViewModel retrieveViewModel) {
```
Comment (ryotahidaka) [inline src/java/app/LoginUseCaseFactory.java:32]: Adding retrieve functionality

File: src/java/app/LoginUseCaseFactory.java [hunk 4]
```diff
@@ -27,10 +28,11 @@ public static LoginView create(
...
 
         try {
-            LoginController loginController = createLoginUseCase(viewManagerModel, loginViewModel, loggedInViewModel, userDataAccessObject);
+            LoginController loginController = createLoginUseCase(viewManagerModel, loginViewModel, loggedInViewModel, retrieveViewModel, userDataAccessObject);
```
Comment (ryotahidaka) [inline src/java/app/LoginUseCaseFactory.java:35]: Adding retrieve functionality

File: src/java/app/LoginUseCaseFactory.java [hunk 5]
```diff
@@ -43,10 +45,11 @@ private static LoginController createLoginUseCase(
             ViewManagerModel viewManagerModel,
             LoginViewModel loginViewModel,
             LoggedInViewModel loggedInViewModel,
+            RetrieveViewModel retrieveViewModel,
```
Comment (ryotahidaka) [inline src/java/app/LoginUseCaseFactory.java:48]: Adding retrieve functionality

File: src/java/app/LoginUseCaseFactory.java [hunk 6]
```diff
@@ -43,10 +45,11 @@ private static LoginController createLoginUseCase(
...
             LoginUserDataAccessInterface userDataAccessObject) throws IOException {
 
         // Notice how we pass this method's parameters to the Presenter.
-        LoginOutputBoundary loginOutputBoundary = new LoginPresenter(viewManagerModel, loggedInViewModel, loginViewModel);
+        LoginOutputBoundary loginOutputBoundary = new LoginPresenter(viewManagerModel, loggedInViewModel, loginViewModel, retrieveViewModel);
```
Comment (ryotahidaka) [inline src/java/app/LoginUseCaseFactory.java:52]: Adding retrieve functionality

File: src/java/app/ResultsUseCaseFactory.java [hunk 7]
```diff
@@ -29,7 +29,7 @@ public static ResultsView create(
             ResultsDataAccessInterface userDataAccessObject) {
 
         try {
-            ResultsController resultsController = createResultsUseCase(viewManagerModel, resultsViewModel, userDataAccessObject);
+            ResultsController resultsController = createResultsUseCase(viewManagerModel, resultsViewModel, retrieveViewModel, userDataAccessObject);
```
Comment (ryotahidaka) [inline src/java/app/ResultsUseCaseFactory.java:32]: Adding retrieve functionality

File: src/java/app/ResultsUseCaseFactory.java [hunk 8]
```diff
@@ -41,10 +41,11 @@ public static ResultsView create(
     private static ResultsController createResultsUseCase(
             ViewManagerModel viewManagerModel,
             ResultsViewModel resultsViewModel,
+            RetrieveViewModel retrieveViewModel,
```
Comment (ryotahidaka) [inline src/java/app/ResultsUseCaseFactory.java:44]: Adding retrieve functionality

File: src/java/app/ResultsUseCaseFactory.java [hunk 9]
```diff
@@ -41,10 +41,11 @@ public static ResultsView create(
...
             ResultsDataAccessInterface userDataAccessObject) throws IOException {
 
         // Notice how we pass this method's parameters to the Presenter.
-        ResultsOutputBoundary resultsOutputBoundary = new ResultsPresenter(viewManagerModel, resultsViewModel);
+        ResultsOutputBoundary resultsOutputBoundary = new ResultsPresenter(viewManagerModel, resultsViewModel, retrieveViewModel);
```
Comment (ryotahidaka) [inline src/java/app/ResultsUseCaseFactory.java:48]: Adding retrieve functionality

File: src/java/data_access/FileUserDataAccessObject.java [hunk 10]
```diff
@@ -75,7 +75,7 @@ public FileUserDataAccessObject(String accountPath, String exercisePath, UserFac
                     String name = "";
                     ArrayList<Exercise> exerciselist = new ArrayList<>();
                     for (String i: col){
-                        String[] element = i.split(",");
+                        String[] element = i.split("\\|");
```
Comment (ryotahidaka) [inline src/java/data_access/FileUserDataAccessObject.java:78]: Changed delimiter

File: src/java/data_access/FileUserDataAccessObject.java [hunk 11]
```diff
@@ -135,14 +135,14 @@ private void saveExercise(){
         BufferedWriter writer;
         try {
             writer = new BufferedWriter(new FileWriter(exerciseFile));
-            writer.write("username; exercises(title, muscle, description, difficulty)");
+            writer.write("username; exercises(title| muscle| description| difficulty)");
```
Comment (ryotahidaka) [inline src/java/data_access/FileUserDataAccessObject.java:138]: Changed delimiter

File: src/java/data_access/FileUserDataAccessObject.java [hunk 12]
```diff
@@ -135,14 +135,14 @@ private void saveExercise(){
...
             writer.newLine();
 
             for (String username : exercises.keySet()) {
                 String line = username + ";";
                 ArrayList<Exercise> exerList = exercises.get(username);
                 for (Exercise exer: exerList){
-                    line = line + String.format("%s,%s,%s, %s;", exer.getTitle(), exer.getMuscle(), exer.getDescription(), exer.getDifficulty());
+                    line = line + String.format("%s|%s|%s|%s;", exer.getTitle(), exer.getMuscle(), exer.getDescription(), exer.getDifficulty());
```
Comment (ryotahidaka) [inline src/java/data_access/FileUserDataAccessObject.java:145]: Changed delimiter

**Review by ryotahidaka — COMMENTED** [review]
Should be all running well with new implementation

**Review by DavidFJ207 — APPROVED** [review]

**Pull request merged**